From 97518ca8bda91ce12c503197a98fa71690cb67f9 Mon Sep 17 00:00:00 2001
From: Will Cosgrove <will@panic.com>
Date: Fri, 12 May 2017 16:34:26 -0700
Subject: [PATCH] Fix memory leak of crypt_ctx->h using openSSL 1.1+ (#177)

Need to use EVP_CIPHER_CTX_free instead of EVP_CIPHER_CTX_reset.
---
 src/openssl.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/openssl.h b/src/openssl.h
index 6aa12192..2bd80b96 100644
--- a/src/openssl.h
+++ b/src/openssl.h
@@ -267,7 +267,7 @@ int _libssh2_md5_init(libssh2_md5_ctx *ctx);
 #define _libssh2_cipher_3des EVP_des_ede3_cbc
 
 #ifdef HAVE_OPAQUE_STRUCTS
-#define _libssh2_cipher_dtor(ctx) EVP_CIPHER_CTX_reset(*(ctx))
+#define _libssh2_cipher_dtor(ctx) EVP_CIPHER_CTX_free(*(ctx))
 #else
 #define _libssh2_cipher_dtor(ctx) EVP_CIPHER_CTX_cleanup(ctx)
 #endif
